app/grandchallenge/reader_studies/static/reader_studies/js/remove_hidden_input_fields.js

```diff
@@ -0,0 +1,9 @@
+form_id = document.getElementById("form-id").dataset.formid
+document.querySelector('#form-'+form_id).addEventListener("submit", function(e){
+    // this removes the hidden interface input fields that flexible-image-widget.html and image-search-widget.html add
+    // to prevent multiple interface values from being submitted with the form

```

Comment:
Why filter these out client side rather than server side?

Reply:
Changing the request data server side (in the post() function on the view seemed like a bad idea, the request object is probably immutable for a good reason). Handling the multiple values for the `interface` key in the data further down the line required changes in multiple places in the codebase, and so preventing them from being sent along in the first place seemed best. How exactly would you suggest doing this server side?

Reply:
Doing this client side is hacky. You shouldn't have to rely on the client running something for validation and use. There is nothing to stop a client sending multiple values for the interface now right, so it would need to be handled server side anyway.

Reply:
That makes sense. I have added a server-side solution, but I'm not sure if updating the request data is the best way to go. It's definitely the most straight-forward since it only requires changes in one place, but it does feel a bit hacky as well.

Reply:
Yes, it's still not great but is better to do on the server side. Can't you do this as part of the value validation on the form?

Is the problem using htmx for this? Would a select 2 (multiple) not be better?

Reply:
We're using a select2 for the interface here already, but for the FlexibleImageWidget we use htmx to update the form in place after a user has chosen a subwidget, and this is when the extra interface values get added. 

Value validation on the `clean_field` method doesn't work because that method never gets called because the clean method on the field (which is run before) already fails validation. I don't think you can (or at least couldn't figure out how to) override the clean method of the ModelChoiceField (which is only defined in the form's init method) So the only way (as far as I can see) to fix this is to override the full_clean method and change the form data before any other clean methods are called. But this is essentially what I did before on the post method.

Reply:
Can't you give the field a different name on the sub widget?

Reply:
That will affect all other places where the widget is used. I'll check if it causes similarly complicated issues elsewhere.

Reply:
Changing the name fixes the problem without introducing issues elsewhere. Feeling a bit stupid for not thinking of that solution myself.

Reply:
I mean it took us both a while to get there, it's the new year after all!